src/jit/gentree.cpp

```diff
@@ -18151,6 +18169,43 @@ GenTree* Compiler::gtNewMustThrowException(unsigned helper, var_types type, CORI
     }
     return node;
 }
+
+// Returns true for the HW Instrinsic instructions that have MemoryStore semantics, false otherwise
+bool GenTreeHWIntrinsic::OperIsMemoryStore()
+{
+#ifdef _TARGET_XARCH_
+    // Some xarch instructions have MemoryStore sematics
+    HWIntrinsicCategory category = Compiler::categoryOfHWIntrinsic(gtHWIntrinsicId);
+    if (category == HW_Category_MemoryStore)
+    {
+        return true;
+    }
+    else if ((gtHWIntrinsicId == NI_AVX_ExtractVector128 || gtHWIntrinsicId == NI_AVX2_ExtractVector128) &&
+             Compiler::numArgsOfHWIntrinsic(this) == 3)
+    {
+        // Also check for Avx/Avx2.ExtractVector128 which has the category of HW_Category_IMM.
+        return true;
+    }
+
+#endif // _TARGET_XARCH_
+    return false;
+}
+
+// Returns true for the HW Instrinsic instructions that have MemoryLoad semantics, false otherwise
+bool GenTreeHWIntrinsic::OperIsMemoryLoad()
+{
+#ifdef _TARGET_XARCH_
+    // Some xarch instructions have MemoryLoad sematics
+    HWIntrinsicCategory category = Compiler::categoryOfHWIntrinsic(gtHWIntrinsicId);
+    if (category == HW_Category_MemoryLoad)
```

Comment:
We need to check `Avx/Avx2.InsertVector128` as well that may have memory load overloads.
```cpp
// If the type of the second argument of Avx/Avx2.InsertVector128 is pointer, the intrinsic is a special memory load
if (category == HW_Category_MemoryLoad ||
    ((gtHWIntrinsicId == AVX_ InsertVector128 || gtHWIntrinsicId == AVX2_ InsertVector128) && op1->AsArgList()->Rest()->Current()->TypeGet() == TYP_I_IMP))
```